### PR TITLE
Remove Storage.sol

### PR DESCRIPTION
**Pull request by markspanbroek**
`Storage` was already a thin wrapper around `Marketplace` without a clear responsibility of its own. Its remaining functionality is moved to `Proofs` and `Marketplace`.

### CODE REVIEW

File: contracts/Proofs.sol [hunk 1]
```diff
@@ -38,25 +37,24 @@ contract Proofs {
     return timeout;
   }
 
-  function _end(EndId endId) internal view returns (uint256) {
-    uint256 end = ends[endId];
+  function _end(RequestId request) internal view returns (uint256) {
```
Comment (emizzle) [inline contracts/Proofs.sol:40]: Nitpick: my personal preference would be name the variable `requestId` to highlight its type `RequestId` as opposed to a `Request` object.

File: contracts/Proofs.sol [hunk 2]
```diff
@@ -20,15 +19,15 @@ contract Proofs {
...
-  mapping(ProofId => EndId) private idEnds;
-  mapping(ProofId => uint256) private probabilities;
-  mapping(ProofId => uint256) private markers;
-  mapping(ProofId => uint256) private missed;
-  mapping(ProofId => mapping(uint256 => bool)) private received;
-  mapping(ProofId => mapping(uint256 => bool)) private missing;
+  mapping(SlotId => bool) private ids;
+  mapping(SlotId => uint256) private starts;
+  mapping(RequestId => uint256) private ends;
+  mapping(SlotId => RequestId) private requestIds;
```
Comment (emizzle) [inline contracts/Proofs.sol:25]: Nitpick personal preference, other naming options:
- `slotRequests`
- `slotRequestIds` 
- `requestsForSlot`
It was confusing for me to read `requestIds[slotId]`, but at the end of the day, it's not a problem as the compiler will enforce the right types. More of a personal preference.

File: contracts/Proofs.sol [hunk 3]
```diff
@@ -20,15 +19,15 @@ contract Proofs {
...
-  mapping(ProofId => bool) private ids;
-  mapping(ProofId => uint256) private starts;
-  mapping(EndId => uint256) private ends;
-  mapping(ProofId => EndId) private idEnds;
-  mapping(ProofId => uint256) private probabilities;
-  mapping(ProofId => uint256) private markers;
-  mapping(ProofId => uint256) private missed;
-  mapping(ProofId => mapping(uint256 => bool)) private received;
-  mapping(ProofId => mapping(uint256 => bool)) private missing;
+  mapping(SlotId => bool) private ids;
```
Comment (emizzle) [inline contracts/Proofs.sol:22]: Nitpick personal preference: could we call this `slotIds`?

**Review by emizzle — APPROVED** [review]
LGTM mark, great job! Just some nitpick personal preferences, but nothing blocking at all.

File: contracts/Proofs.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (markspanbroek) [reply contracts/Proofs.sol:22]: Yes, I was thinking along the same lines. It's a tough choice, but eventually the improved readability won over the decoupling for me. I don't think we'll reuse the proof system anywhere else, but should we want to, we can re-introduce `ProofId` and `EndId`.

File: contracts/Proofs.sol (hunk 1 above)
Comment (markspanbroek) [reply contracts/Proofs.sol:40]: Agreed, I'll fix this.

File: contracts/Proofs.sol (hunk 2 above)
Comment (markspanbroek) [reply contracts/Proofs.sol:25]: Agreed. I'll change it to `requestForSlot`.

File: contracts/Proofs.sol (hunk 3 above)
Comment (markspanbroek) [reply contracts/Proofs.sol:22]: Yes, I'll change it.

**Pull request merged**